io-engine/src/bdev/nexus/nexus_io.rs

```diff
@@ -185,6 +175,12 @@ impl<'n> NexusBio<'n> {
 
     /// TODO
     pub(super) fn submit_request(mut self) {
+        if self.channel().is_frozen() {
+            let s = self.clone();
+            self.channel_mut().freeze_io_submission(s);
```

Comment:
Have we run any heavy workload to see how big we can grow. I'm thinking of a condition where the freeze list is huge and keeps holding mempool objects, leading to failure in bio allocation for new IOs.

Reply:
Nexus subsystem is perhaps paused at this point so the limit would be bound by how many IOs are already in the system.

Reply:
It's actually not paused, so we will be receiving IOs...
This may be a problem indeed... though on the usual case where etcd it shouldn't be an issue, and if etcd is down we now have timeouts to give up.
Ah @dsavitskiy, when we give up and shutdown the nexus, do we free the IO's?

Reply:
@tiagolobocastro  When we give up writing to ETCD, we fail and drop all frozen I/Os.

We should probably create a separate ticket to handle ENOMEM during I/O submission properly, as this is not the only place when ENOMEM can happen. SPDK itself does have some ENOMEM-during-submit processing logic.

Reply:
@dsharma-dc @tiagolobocastro I create an epic to cover ENOMEM issues: https://dcsw-bolt.atlassian.net/browse/GTM-997.

Reply:
I'd suggest we quantify our vulnerability for ENOMEM under this scenario, and if we don't happen to run into this under a minimally/moderately loaded system then it can be addressed under this new epic.